spec/models/atmosphere/user_fund_spec.rb

```diff
@@ -33,4 +33,75 @@
         to include(I18n.t('funds.unique_user'))
   end
 
+
+  describe '#ensure_single_default' do
+
+    it 'makes first user_fund default' do
+      expect(user_fund.default).to be_truthy
+    end
+
+    it 'blocks setting the only fund as not default' do
+      expect(user_fund.default).to be_truthy
+      user_fund.default = false
+      user_fund.save
+      expect(user_fund.reload.default).to be_truthy
+    end
+
+    it 'does not bother single default fund creation' do
+      pc = create(:poor_chap)
+      uf = create(:user_fund, user: pc, default: true)
+      expect(uf.reload.default).to be_truthy
+    end
+
+    it 'keeps only single default user fund' do
+      uf = create(:user_fund, user: user_fund.user, default: true)
+      expect(Atmosphere::UserFund.count).to eq 2
```

Comment:
Sure. Unless the user factory (and you need a user there) creates two different funds just for the fun of it! So I end up spending a better part of an hour figuring these things out. Then I switched to my better :poor_chap factory but I decided to those leave "theoretically stupid" (but, in fact, quite useful) expectations in the code.